torch/_inductor/ir.py

```diff
@@ -4108,21 +4119,40 @@ def codegen(self, wrapper):
             f"{output_name}_pg = _find_or_create_pg_by_ranks_and_tag('{tag}', {ranks}, {group_size})"
         )
 
+        self.codegen_output(wrapper, output_name, input_names)
         self.codegen_collective(wrapper, output_name, input_names)
-
         wrapper.writeline(f"_register_tensor_work({output_name}, {output_name}_work)")
 
 
-class MultiOutputNoSizeAssert(MultiOutput):
-    """
-    Extract partial output from a multi-output OP.
-        Works like MultiOutput but doesn't assert size. This must be a property guaranteed by the op emiting this.
-    """
+class InPlaceCollectiveKernel(CollectiveKernel):
```

Comment:
can we add some documentation about what's a `InPlaceCollectiveKernel` and `OutPlaceCollectiveKernel`, and what's their difference? It would also be beneficial to explain when to extend which kernel so that new collectives adding to inductor can have some guidance which to extend.

Reply:
Good point. I added documentation to those two